controller/sync_namespace.go

```diff
@@ -1,46 +1,50 @@
 )
 
-func syncNamespace(resourceTracking argo.ResourceTracking, appLabelKey string, trackingMethod v1alpha1.TrackingMethod, appName string, syncPolicy *v1alpha1.SyncPolicy) func(un *unstructured.Unstructured) (bool, error) {
-	return func(liveNs *unstructured.Unstructured) (bool, error) {
-		if liveNs != nil && kube.GetAppInstanceLabel(liveNs, cdcommon.LabelKeyAppInstance) != "" {
-			kube.UnsetLabel(liveNs, cdcommon.LabelKeyAppInstance)
-			return true, nil
+// syncNamespace determine if Argo CD should create and/or manage the namespace
+// where the application will be deployed.
+func syncNamespace(resourceTracking argo.ResourceTracking, appLabelKey string, trackingMethod v1alpha1.TrackingMethod, appName string, syncPolicy *v1alpha1.SyncPolicy) func(m, l *unstructured.Unstructured) (bool, error) {
+	// This function must return true for the managed namespace to be synced.
+	return func(managedNs, liveNs *unstructured.Unstructured) (bool, error) {
+		if managedNs == nil {
+			return false, nil
 		}
 
-		isNewNamespace := liveNs != nil && liveNs.GetUID() == "" && liveNs.GetResourceVersion() == ""
+		isNewNamespace := liveNs == nil
+		isManagedNamespace := syncPolicy != nil && syncPolicy.ManagedNamespaceMetadata != nil
 
-		if liveNs != nil && syncPolicy != nil {
-			// managedNamespaceMetadata relies on SSA, and since the diffs are computed by the k8s control plane we
-			// always need to call the k8s api server, so we'll always need to return true if managedNamespaceMetadata is set.
-			hasManagedMetadata := syncPolicy.ManagedNamespaceMetadata != nil
-			if hasManagedMetadata {
-				managedNamespaceMetadata := syncPolicy.ManagedNamespaceMetadata
-				liveNs.SetLabels(managedNamespaceMetadata.Labels)
-				liveNs.SetAnnotations(appendSSAAnnotation(managedNamespaceMetadata.Annotations))
-
-				err := resourceTracking.SetAppInstance(liveNs, appLabelKey, appName, "", trackingMethod)
-				if err != nil {
-					return false, fmt.Errorf("failed to set app instance tracking on the namespace %s: %s", liveNs.GetName(), err)
-				}
+		// should only sync the namespace if it doesn't exist in k8s or if
+		// syncPolicy is defined to manage the metadata
+		if !isManagedNamespace && !isNewNamespace {
+			return false, nil
+		}
 
-				return true, nil
-			}
+		if isManagedNamespace {
+			managedNamespaceMetadata := syncPolicy.ManagedNamespaceMetadata
+			managedNs.SetLabels(managedNamespaceMetadata.Labels)
+			// managedNamespaceMetadata relies on SSA in order to avoid overriding
+			// existing labels and annotations in namespaces
+			managedNs.SetAnnotations(appendSSAAnnotation(managedNamespaceMetadata.Annotations))
 		}
 
-		return isNewNamespace, nil
+		// TODO: https://github.com/argoproj/argo-cd/issues/11196
+		// err := resourceTracking.SetAppInstance(managedNs, appLabelKey, appName, "", trackingMethod)
+		// if err != nil {
+		// 	return false, fmt.Errorf("failed to set app instance tracking on the namespace %s: %s", managedNs.GetName(), err)
+		// }
```

Comment:
Usually I don't do this but I left this intentionally committed so the person who works on #11196 has a partial implementation ready.

Reply:
Would it be sufficient to pass in an instance of `appStateManager` and to then do a pre-check with `appStateManager.isSelfReferencedObj`?  So for this to work as discussed in #11196, in order to return an error we'd do something like below? 

```suggestion
		selfReferencedObj := appStateManager.isSelfReferencedObj(liveNs, managedNs, appName, appLabelKey, trackingMethod)
		if !selfReferencedObj {
			return false, fmt.Errorf("insert error here")
		}

		err := resourceTracking.SetAppInstance(managedNs, appLabelKey, appName, "", trackingMethod)
		if err != nil {
			return false, fmt.Errorf("failed to set app instance tracking on the namespace %s: %s", managedNs.GetName(), err)
		}
```

Reply:
@blakepettersson Exactly! This would address this conflict validation logic that we discussed in #11196 . 
But I'm keeping this outside of this PR as I'm fixing another issue here.